plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveLocal.java

```diff
@@ -126,4 +161,139 @@ public void testGetTableSchemaOffline()
+    {
+        try (Transaction transaction = newTransaction()) {
+            ConnectorSession session = newSession();
+
+            String tableOwner = session.getUser();
+            String schemaName = schemaTableName.getSchemaName();
+            String tableName = schemaTableName.getTableName();
+
+            Table.Builder tableBuilder = Table.builder()
+                    .setDatabaseName(schemaName)
+                    .setTableName(tableName)
+                    .setOwner(tableOwner)
+                    .setTableType(TableType.EXTERNAL_TABLE.name())
+                    .setParameters(ImmutableMap.of(
+                            PRESTO_VERSION_NAME, TEST_SERVER_VERSION,
+                            PRESTO_QUERY_ID_NAME, session.getQueryId()))
+                    .setDataColumns(columns)
+                    .setPartitionColumns(partitionColumns);
+
+            tableBuilder.getStorageBuilder()
+                    .setLocation(externalLocation.toString())
+                    .setStorageFormat(StorageFormat.create(hiveStorageFormat.getSerDe(), hiveStorageFormat.getInputFormat(), hiveStorageFormat.getOutputFormat()))
+                    .setBucketProperty(bucketProperty)
+                    .setSerdeParameters(ImmutableMap.of());
+
+            PrincipalPrivileges principalPrivileges = testingPrincipalPrivilege(tableOwner, session.getUser());
+            transaction.getMetastore().createTable(session, tableBuilder.build(), principalPrivileges, Optional.of(externalLocation), true, EMPTY_TABLE_STATISTICS);
+
+            transaction.commit();
+        }
+    }
+
+    private java.nio.file.Path extractResourceZipToTemporaryDirectory(String resourceName)
+            throws IOException
+    {
+        java.nio.file.Path tempDir = java.nio.file.Files.createTempDirectory(getClass().getSimpleName()).normalize();
+        log.info(format("Extracting resource '%s' to %s", resourceName, tempDir));
+        InputStream is = getClass().getClassLoader().getResourceAsStream(resourceName);
+        ZipInputStream zis = new ZipInputStream(is);
+        byte[] buffer = new byte[1024];
+        ZipEntry entry;
+        while ((entry = zis.getNextEntry()) != null) {
+            java.nio.file.Path extractedEntryPath = tempDir.resolve(entry.getName()).normalize();
+            if (!extractedEntryPath.startsWith(tempDir)) {
+                throw new IOException("ZIP contents escapes destination directory");
+            }
+            if (entry.isDirectory()) {
+                java.nio.file.Files.createDirectories(extractedEntryPath);
+            }
+            else {
+                try (OutputStream os = java.nio.file.Files.newOutputStream(extractedEntryPath)) {
+                    int len;
+                    while ((len = zis.read(buffer)) > 0) {
+                        os.write(buffer, 0, len);
+                    }
+                }
+            }
+        }
+        return tempDir;
+    }
```

Comment:
What's the `spark_bucketed_nation.zip` being added in this PR? looks empty

Reply:
Looks like a mistake, I'll check.

Reply:
It looks like GitHub is reporting the wrong file size for `spark_bucketed_nation.zip`, the actual file is 4609 bytes, and contains a zipped directory with Spark-generated buckets.

Reply:
Why would we want to use a zip for these?

Reply:
Agreed, can we simply add them as resources?

Reply:
In one of the earlier iterations I've tried running the test with external table location pointing to a resource dir. It worked fine locally when I had sources checked out, but failed on CI reporting directory not found. The path pointed inside a JAR file. I could change the ZIP file into a resource dir, but I'd still need to extract the files from the JAR to some temporary directory, and this seems easier when just one ZIP file is involved.

Reply:
I think it's cleaner to explicitly copy the resources to temporary files, rather than unzip a zip file, since that makes it clear what is going on (otherwise you need to go look in the zip file).

Reply:
> I think it's cleaner to explicitly copy the resources to temporary files, rather than unzip a zip file, since that makes it clear what is going on (otherwise you need to go look in the zip file).

Sure, I changed the setup to use a resource directory instead of a zip file.